HotelBooking/HotelManager.cs

```diff
@@ -1,47 +1,43 @@
 ﻿using System;
-using HotelExceptions;
 
 namespace HotelBooking
 {
-    public class HotelManager : IHotelReservation
+    public class HotelManager
     {
         private readonly IHotelFactory _factory;
         private readonly IPaymentService _paymentService;
         private readonly IBookingService _bookingService;
+        private readonly ILogger _logger;
 
-        public IHotel Hotel { get; set; }
-        public HotelManager(IHotelFactory factory, IPaymentService paymentService, IBookingService bookingService)
+        private IHotel _hotel;
+
+        public HotelManager(IHotelFactory factory, IPaymentService paymentService, IBookingService bookingService,
+            ILogger logger)
         {
             _factory = factory;
             _paymentService = paymentService;
             _bookingService = bookingService;
-
+            _logger = logger;
         }
 
         public ReservationResult MakeReservation(int hotelId, double price, int creditCardNumber,
             string email, DateTime date)
         {
-            try
-            {
-                FindHotel(hotelId);
-            }
-            catch (NullHotelException e)
+            bool hotelFound = FindHotel(hotelId);
+
+            if (!hotelFound)
             {
-                Console.WriteLine(e.Message);
-                return new ReservationResult( false);
+                _logger.Write($"Couldn't find hotel with ID: {hotelId}");
+                return new ReservationResult(false);
             }
 
-            return Hotel.Reserve(date, price, creditCardNumber, email);
-
+            return _hotel.Reserve(date, price, creditCardNumber, email);
         }
 
-        internal void FindHotel(int hotelId)
+        private bool FindHotel(int hotelId)
```

Comment:
Nice to have:

I think that the good practice is to follow some already existing name conventions. In this case I would expect that FindHotel method will return to me hotel or null. If you want method that just check if hotel exist you should name it Exist - I would advise to take a closer look for MS linq names.